### DISCUSSION TITLE
If we could change the size of grid in ui.scene()?

### DISCUSSION THREAD

**Discussion by v479038280** [zauberzeug/nicegui#3254] [Q&A] [answered]
### Question

It seems the default size of the grid(the red area) is 100*100,if we could change its value?
![微信截图_20240620183934](https://github.com/zauberzeug/nicegui/assets/25096045/81415515-2d69-4a14-b4f4-f8540f569f06)

**Comment by python-and-novella** [comment]
in `.venv\Lib\site-packages\nicegui\elements\scene.js`,Line 135
you will find grid's init function:
```javascript
const grid = new THREE.GridHelper(100, 100);
```
you could modify it.

**Reply by v479038280** [reply]
I have tried this way before, but it did not work.

**Reply by python-and-novella** [reply]
For me,it works.
![image](https://github.com/zauberzeug/nicegui/assets/16256910/463e226e-fc34-4c4d-89a7-d488b7d6b66f)
Maybe you do it in wrong file.

**Reply by v479038280** [reply]
I change it to(10,20),but it's still 100*100
![微信截图_20240621111318](https://github.com/zauberzeug/nicegui/assets/25096045/d2853c29-a70e-4485-a3a0-5fa47a8a88ae)


![微信截图_20240621111501](https://github.com/zauberzeug/nicegui/assets/25096045/74d50837-4ac6-43c8-a3f4-f2c6aa074a97)

**Reply by python-and-novella** [reply]
If your program is executed in venv,you should modify it which in `$Project_Root_Path$\.venv\Lib\site-packages\nicegui\elements\scene.js`
Otherwise,it's python Interpreter‘s `Lib\site-packages\nicegui\elements\scene.js`

**Reply by v479038280** [reply]
I got some strange result:
First I change it to (10,20),I got this with microsoft Edge Browser:
![111](https://github.com/zauberzeug/nicegui/assets/25096045/781301f7-953a-4b3e-86dc-de5522a8460f)

Then I changed it to(5,10),the result is still above picture.

When I use another browser the result is always the same (100,100):
![11111](https://github.com/zauberzeug/nicegui/assets/25096045/e7461b71-73f8-4259-88cf-23425ac39a93)

**Reply by v479038280** [reply]
> I got some strange result: First I change it to (10,20),I got this with microsoft Edge Browser: ![111](https://private-user-images.githubusercontent.com/25096045/341631517-781301f7-953a-4b3e-86dc-de5522a8460f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTg5NDE2MjAsIm5iZiI6MTcxODk0MTMyMCwicGF0aCI6Ii8yNTA5NjA0NS8zNDE2MzE1MTctNzgxMzAxZjctOTUzYS00YjNlLTg2ZGMtZGU1NTIyYTg0NjBmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjIxVDAzNDIwMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTVkNGM2NmYxYThiMTEzOTgwNTk1ZTcwYjkxZDZhM2Y4YjQ3NWMzOGYyMDUzNDU2NjAzYTFmNWQ3YTk4ZTA1YjgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.vaiv08xxS-C-etWJu2Ip45acxvMDg8sYFA9LyuemNj8)
> 
> Then I changed it to(5,10),the result is still above picture.
> 
> When I use another browser the result is always the same (100,100): ![11111](https://private-user-images.githubusercontent.com/25096045/341632128-e7461b71-73f8-4259-88cf-23425ac39a93.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTg5NDE2MjAsIm5iZiI6MTcxODk0MTMyMCwicGF0aCI6Ii8yNTA5NjA0NS8zNDE2MzIxMjgtZTc0NjFiNzEtNzNmOC00MjU5LTg4Y2YtMjM0MjVhYzM5YTkzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjIxVDAzNDIwMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTVlMDY1MjQ0ZjExZDA2ZGIzM2I4M2FmZWRlNGRhOTQ2NTY5NDA4MGJjMzJlMGMwNTkyMDY1NDZkODU5NmU0YjMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.hw_eCgeSp9fjO5_4LcBsp8hHTpakhEh6sPlmYlCtQfI)

Then I restart both browers, and use the value(5,10):
here is the result
![333](https://github.com/zauberzeug/nicegui/assets/25096045/37d7cbf1-9413-46d6-b385-17a1ce75e93c)
then I cleared the catch file in edge ,the result is still above

**Reply by python-and-novella** [reply]
you need restart your program entirely .this method didn't support reload yet.you can only modify it manually before you run program.

**Reply by python-and-novella** [reply]
it's browser's cache,plz refresh forcely

**Reply by v479038280** [reply]
Yes,we must clear the cache manually everytime . 
Now I set it to(2*4),   the grid is always square ,not  a 2X4 rectangle

![111](https://github.com/zauberzeug/nicegui/assets/25096045/6877a9a9-353f-40cf-a237-b0eae00aa6f3)

**Reply by python-and-novella** [reply, marked as answer]
parameter is size and divisions,not width and height.
More info can be found in threejs.org.

**Comment by rodja** [comment]
It's not configurable right now. We are happy to consider a pull request to provide this option.

**Reply by falkoschindler** [reply]
I just created PR #3257.
